purchase_request/wizard/purchase_request_line_make_purchase_order.py

```diff
@@ -0,0 +1,325 @@
+    purchase_order_id = fields.Many2one('purchase.order',
+                                        string='Purchase Order',
+                                        domain=[('state', '=', 'draft')])
+    sync_data_planned = fields.Boolean(
+        string="Merge on PO lines with equal Scheduled Date")
+
+    @api.model
+    def _prepare_item(self, line):
+        return {
+            'line_id': line.id,
+            'request_id': line.request_id.id,
+            'product_id': line.product_id.id,
+            'name': line.name or line.product_id.name,
+            'product_qty': line.product_qty,
+            'product_uom_id': line.product_uom_id.id,
+        }
+
+    @api.model
+    def _check_valid_request_line(self, request_line_ids):
+        picking_type = False
+        company_id = False
+
+        for line in self.env['purchase.request.line'].browse(request_line_ids):
+
+            if line.request_id.state != 'approved':
+                raise UserError(
+                    _('Purchase Request %s is not approved') %
+                    line.request_id.name)
+
+            if line.purchase_state == 'done':
+                raise UserError(
+                    _('The purchase has already been completed.'))
+
+            line_company_id = line.company_id \
+                and line.company_id.id or False
+            if company_id is not False \
+                    and line_company_id != company_id:
+                raise UserError(
+                    _('You have to select lines '
+                      'from the same company.'))
+            else:
+                company_id = line_company_id
+
+            line_picking_type = line.request_id.picking_type_id or False
+            if not line_picking_type:
+                raise UserError(
+                    _('You have to enter a Picking Type.'))
+            if picking_type is not False \
+                    and line_picking_type != picking_type:
+                raise UserError(
+                    _('You have to select lines '
+                      'from the same Picking Type.'))
+            else:
+                picking_type = line_picking_type
+
+    @api.model
+    def default_get(self, fields):
+        res = super(PurchaseRequestLineMakePurchaseOrder, self).default_get(
+            fields)
+        request_line_obj = self.env['purchase.request.line']
+        request_line_ids = self.env.context.get('active_ids', False)
+        active_model = self.env.context.get('active_model', False)
+        if not request_line_ids:
+            return res
+        assert active_model == 'purchase.request.line', \
+            'Bad context propagation'
+        items = []
+        groups = []
+        self._check_valid_request_line(request_line_ids)
+        request_lines = request_line_obj.browse(request_line_ids)
+        for line in request_lines:
+            items.append([0, 0, self._prepare_item(line)])
+            groups.append(line.request_id.group_id.id)
```

Comment:
Could we get rid of this limitation, as normally users create one PO for each supplier indistinctly of PG?
@jbeficent @aheficent @lreficent